### PR TITLE
remove c++ template from cses-1757, add python sol

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/gold/cses-1757.mdx [hunk 1]
```diff
@@ -20,230 +20,44 @@ course in "Course Schedule II." So it suffices to solve the CF problem
 #include <bits/stdc++.h>
 using namespace std;
 
-using ll = long long;
-using ld = long double;
-using db = double;
-using str = string; // yay python!
+int n, m, out[200005]; // Number of outgoing nodes
+vector<int> radj[200005]; // Reverse graph
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:24]: two spaces

Comment (SansPapyrus683) [reply solutions/gold/cses-1757.mdx:24]: also some of these don't have to be global
or all of them, no functions here

File: solutions/gold/cses-1757.mdx [hunk 2]
```diff
@@ -313,4 +127,40 @@ public class Main {
 ```
 </JavaSection>
 
+<PySection>
+```py
+import heapq
+
+n, m = map(int, input().split())
+out = [0] * (200005)  # Number of outgoing nodes
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:135]: magic number
why not dynamically size based on n/m

File: solutions/gold/cses-1757.mdx [hunk 3]
```diff
@@ -313,4 +127,40 @@ public class Main {
...
+	x = -heapq.heappop(pq)  # Remove the node with the greatest id.
+	ans.append(x)
+	for t in radj[x]:
+		out[t] -= 1
+		if out[t] == 0:
+			heapq.heappush(pq, -t)  # Remove all edges that begin at `x`.
+
+ans = ans[::-1]
+
+for t in ans:
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:161]:
```suggestion
for t in reversed(ans):
```

File: solutions/gold/cses-1757.mdx [hunk 4]
```diff
@@ -313,4 +127,40 @@ public class Main {
...
+
+for _ in range(m):
+	a, b = map(int, input().split())
+	radj[b].append(a)
+	out[a] += 1
+
+ans = []
+pq = []
+
+for i in range(1, n+1):
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:146]: spaces

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/cses-1757.mdx [hunk 5]
```diff
@@ -313,4 +129,38 @@ public class Main {
...
+while pq:
+	x = -heapq.heappop(pq)  # Remove the node with the greatest id.
+	ans.append(x)
+	for t in radj[x]:
+		out[t] -= 1
+		if out[t] == 0:
+			heapq.heappush(pq, -t)  # Remove all edges that begin at `x`.
+
+for t in reversed(ans):
+	print(t, end=' ')
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:162]:
```suggestion
print(' '.join(str(i) for i in reversed(ans)))
```

File: solutions/gold/cses-1757.mdx [hunk 6]
```diff
@@ -313,4 +129,38 @@ public class Main {
...
+	a, b = map(int, input().split())
+	radj[b].append(a)
+	out[a] += 1
+
+ans = []
+pq = []
+
+for i in range(1, n + 1):
+	if out[i] == 0:
+		# Any node with out[i] == 0 can be used, so we store all possible nodes in a max heap to get the node with the maximum id.
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:150]: over 80?

Comment (876pol) [reply solutions/gold/cses-1757.mdx]: oops, fixed

File: solutions/gold/cses-1757.mdx [hunk 7]
```diff
@@ -147,7 +147,8 @@ pq = []
 
 for i in range(1, n + 1):
 	if out[i] == 0:
-		# Any node with out[i] == 0 can be used, so we store all possible nodes in a max heap to get the node with the maximum id.
+		"""Any node with out[i] == 0 can be used, so we store all possible
+		nodes in a max heap to get the node with the maximum id."""
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:151]:
```suggestion
		# Any node with out[i] == 0 can be used, so we store all possible
		# nodes in a max heap to get the node with the maximum id.
```

File: solutions/gold/cses-1757.mdx [hunk 8]
```diff
@@ -20,230 +20,46 @@ course in "Course Schedule II." So it suffices to solve the CF problem
...
-	// ex. try to read letter into int
-	if (sz(s)) setIn(s+".in"), setOut(s+".out"); // for USACO
-}
+	/*
+	 * Any node with out[i] == 0 can be used, so we store all possible
+	 * nodes in a max heap to get the node with the maximum id.
+	 */
+	priority_queue<int> pq;
+	for (int i = 1; i <= n; i++) {
+		if (out[i] == 0) pq.push(i);
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:43]: braces

File: solutions/gold/cses-1757.mdx [hunk 9]
```diff
@@ -313,4 +129,37 @@ public class Main {
...
+		heapq.heappush(pq, -i)
+
+while pq:
+	x = -heapq.heappop(pq)  # Remove the node with the greatest id.
+	ans.append(x)
+	for t in radj[x]:
+		out[t] -= 1
+		if out[t] == 0:
+			heapq.heappush(pq, -t)  # Remove all edges that begin at `x`.
+print(' '.join(str(i) for i in reversed(ans)))
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1757.mdx:161]:
```suggestion

print(' '.join(str(i) for i in reversed(ans)))
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**